### DISCUSSION TITLE
Lesson 9: .env / {PRIVATE_KEY}

### DISCUSSION THREAD

**Discussion by SquilliamX** [Cyfrin/foundry-full-course-cu#607] [Q&A] [answered]
Hi I am up to 6:30:19 and he has a hex number. I was wondering where he got that hex number from? It's not a private key (even though it says in line 2 ``PRIVATE_KEY=7ac3722..") because it doesnt start in 0x and I can't find where he added it. What is this hex value and where did he get it from? I assume it is like the $SEPOLIA_RPC_URL, where everyone has their own url

**Comment by shikhar229169** [comment]
Actually that's a private key. It will work both ways with or without the prefix 0x
If you want the private key for anvil network, just run ```anvil``` command in the terminal and there will be a list of private keys (having fake eth).

And to get private key from metamask wallet.
Open metamask, on the top you will se Account, click on it. Then click the three dots corresponding to the account you want to get private key, then click on account details and then show private key, enter your metamask password to get the private key.

**Reply by SquilliamX** [reply]
I see, thank you. Is the private key in the video the anvil private key or the testing wallet private key?

**Reply by theirrationalone** [reply, marked as answer]
PRIVATE_KEY=YOUR_METAMASK_KEY
ANVIL_KEY=ANVIL_PRIVATE_KEY

Both are separated by their name!
